src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -149,6 +149,28 @@ sealed abstract class ArraySeq[+A]
     case _ =>
       super.equals(other)
   }
+
+  override def sorted[B >: A](implicit ord: Ordering[B]): ArraySeq[A] = {
+    val len = this.length
+    if (len <= 1) {
+      this
+    } else {
+      val arr = unsafeArray match {
+        case a: Array[AnyRef] =>
+          a.clone()
+        case _ =>
+          val xs = new Array[AnyRef](len)
```

Comment:
Can this array allocation and subsequent loop be replaced by `val xs = java.util.Arrays.copyOf(unsafeArray, len, classOf[Array[AnyRef]]) `?

Reply:
I think can't use `Arrays.copyOf`

```
scala> val x = Array(1, 2, 3)
x: Array[Int] = Array(1, 2, 3)

scala> java.util.Arrays.copyOf(x, x.length, classOf[Array[AnyRef]])
                               ^
       error: type mismatch;
        found   : Array[Int]
        required: Array[? with Object]
       Note: Int >: ? with Object, but class Array is invariant in type T.
       You may wish to investigate a wildcard type such as `_ >: ? with Object`. (SLS 3.2.10)

scala> java.util.Arrays.copyOf(x.asInstanceOf[Array[AnyRef]], x.length, classOf[Array[AnyRef]])
java.lang.ClassCastException: [I cannot be cast to [Ljava.lang.Object;
  ... 30 elided
```

Reply:
`ArraySeq` is not specialized collection, isn't it? 

How it can happen to have an array with non-boxed primitives in `underlyingArray`?

Reply:
https://github.com/scala/scala/blob/eaeda754e23da72588f9d871254cdd7426d04dea/src/library/scala/collection/immutable/ArraySeq.scala#L278

Reply:
`scala.Array.copyAs` implements all the different cases that are necessary for copying between arbitrary array types.